core/pom.xml

```diff
@@ -22,6 +22,11 @@
       <artifactId>semantic-metrics-api</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.tdunning</groupId>
+      <artifactId>t-digest</artifactId>
```

Comment:
As far as I can tell, this library is not threadsafe, and the usages of its method in this codebase does not wrap it in any concurrency primitives.

Are you sure this is a good fit for highly concurrent usages?